includes/classes/SiteAutomation/Request.php

```diff
@@ -254,18 +273,33 @@ private function request( $timeout ) {
 	/**
 	 * Process response from Sophi.
 	 *
-	 * @param array $response Response of Site Automation API.
-	 * @param bool  $bypass_cache Whether to bypass cache or not.
+	 * @param array    $response Response of Site Automation API.
+	 * @param bool     $bypass_cache Whether to bypass cache or not.
+	 * @param int|bool $post_id The post id to update the post meta with response or false.
 	 *
 	 * @return array
 	 */
-	private function process( $response, $bypass_cache ) {
+	private function process( $response, $bypass_cache, $post_id ) {
 		if ( ! $response ) {
 			return [];
 		}
 
-		if ( ! $bypass_cache ) {
-			update_option( "sophi_site_automation_data_{$this->page}_{$this->widget}", $response );
+		if ( ! $bypass_cache && post_type_exists('sophi-response') ) {
+			if ( $post_id ) {
+				update_post_meta( $post_id, "sophi_site_automation_data_{$this->page}_{$this->widget}", $response );
+				update_post_meta( $post_id, 'sophi_site_automation_last_updated', time() );
+			} else {
+				$post_id = wp_insert_post(
+					[
+						'post_type'  => 'sophi-response',
+						'post_title' => "sophi_site_automation_data_{$this->page}_{$this->widget}",
+					]
+				);
+				if ( $post_id ) {
+					update_post_meta( $post_id, "sophi_site_automation_data_{$this->page}_{$this->widget}", $response );
+					update_post_meta( $post_id, 'sophi_site_automation_last_updated', time() );
+				}
+			}
```

Comment:
Realizing this is much more of a personal preference than anything, can't we check for `! $post_id` and create the post, and only then call the update_post_meta lines (only once)?

Reply:
I think that simplifies the logic and makes it more readable, I've addressed it here: https://github.com/globeandmail/sophi-for-wordpress/pull/298/commits/bd0b247c5de80d7723ba9e6bbc9ac536be0b1dd3